flang/lib/Lower/IntrinsicCall.cpp

```diff
@@ -562,6 +568,12 @@ using I = IntrinsicLibrary;
 static constexpr IntrinsicHandler handlers[]{
     {"abs", &I::genAbs},
     {"achar", &I::genChar},
+    {"adjustl", &I::genAdjustRtCall<Fortran::lower::genAdjustL>,
+      {{{"string", asAddr}}},   
+      /*isElemental=*/true},
+    {"adjustr", &I::genAdjustRtCall<Fortran::lower::genAdjustR>,
```

Comment:
Slick use of templates here. :)

Reply:
@jeanPerier can take credit for that!  :-)